### PR TITLE
Improve logrotate use

### PR DESCRIPTION
**Pull request by aermakov-zalando**
Currently, there are two cronjobs for log rotation, one hourly and one daily. Both try to cleanup the state file and then run `logrotate`. This means that the rotation is only performed once an hour at most, and it's easy to fill up the instance if you log too much, leading to issues such as #176.

This PR adds two new configuration options:

 * `logrotate_interval_minutes` allows running `logrotate` every N minutes instead of hourly. Optional; old behaviour would be preserved if it's unset (run every hour).
 * `rsyslog_application_hardlimit` allows setting a hard limit on the size of `application.log`. If the limit is exceeded, `rsyslogd` would invoke `logrotate` manually before proceeding, and if that doesn't help it'll simply stop writing new messages to that logfile. Applications should try to avoid this, since they'll become blocked until `logrotate` finishes. Optional; off by default.

### CODE REVIEW

**Comment by hjacobs** [comment]
Great, thanks for the work --- I'll let somebody review it more closely.

File: runtime/etc/cron.hourly/logrotate-cleanup [hunk 1]
```diff
@@ -0,0 +1,15 @@
+#!/bin/bash
+(
+    set -e
+    flock 200
```
[the remark this answers is not in the document]
Comment (aermakov-zalando) [reply runtime/etc/cron.hourly/logrotate-cleanup]: It's the FD number of an open file created in the redirect a few lines later. This is a standard idiom for doing locking in shell scripts, see the example in [the man page](https://linux.die.net/man/1/flock).

**Comment by tuxlife** [comment]
:+1:

**Comment by aermakov-zalando** [comment]
👍

**Pull request merged**